app/dashboard/views.py

```diff
@@ -899,11 +899,74 @@ def users_fetch_filters(profile_list, skills, bounties_completed, leaderboard_ra
             organizations__contains=[organisation.lower()]
         )
         profile_list = (profile_list1 | profile_list2).distinct()
-
+    if hackathon_id:
+        profile_list = profile_list.filter(
+            hackathon_registration__hackathon=hackathon_id
+        )
     return profile_list
 
 
 
+@require_GET
+def projects_fetch(request):
+    q = clean(request.GET.get('search', ''), strip=True)
+    order_by = clean(request.GET.get('order_by', '-created_on'), strip=True)
+    skills = clean(request.GET.get('skills', ''), strip=True)
+    filters = clean(request.GET.get('filters', ''), strip=True)
+    sponsor = clean(request.GET.get('sponsor', ''), strip=True)
+    page = clean(request.GET.get('page', 1))
+    hackathon_id = clean(request.GET.get('hackathon', ''))
+
+    try:
+        hackathon_event = HackathonEvent.objects.get(id=hackathon_id)
+    except HackathonEvent.DoesNotExist:
+        hackathon_event = HackathonEvent.objects.last()
+
+    projects = HackathonProject.objects.filter(hackathon=hackathon_event).exclude(status='invalid').prefetch_related('profiles', 'bounty').order_by(order_by)
+
+    if q:
+        projects = projects.filter(
+            Q(name__icontains=q) |
+            Q(summary__icontains=q) |
+            Q(profiles__handle__icontains=q)
+        )
+
+    if sponsor:
+        print(sponsor)
+        projects = projects.filter(
+            Q(bounty__github_url__icontains=sponsor)
+        )
+
+    if skills:
+        projects = projects.filter(
+            Q(profiles__keywords__icontains=skills)
+        )
+
+    if filters == 'winners':
+        projects = projects.filter(
+            Q(badge__isnull=False)
+        )
+
+    projects_paginator = Paginator(projects, 9)
```

Comment:
Lets remove the Paginator object, then just return `has_next = False`, `count = len(projects_data)`, `num_pages = 1` and leave a note to deal with pagination later